skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -0,0 +1,596 @@
+import numpy as np
+from skimage import util
+cimport numpy as np
+cimport cython
+from libc.math cimport exp
+
+ctypedef np.float32_t DTYPE_t
+
+cdef eps = 1.e-8
+
+
+@cython.boundscheck(False)
+cdef inline float patch_distance_2d(DTYPE_t [:, :] p1,
+                                    DTYPE_t [:, :] p2,
+                                    DTYPE_t [:, ::] w, int s):
```

Comment:
Could you add a numpy-style docstring to this? (you can refer to this one in the other similar functions) I can't stand reading non-docstring-ed functions anymore. I want to know what the function does and what each of the inputs is, especially with single-char names like `w` and `s`.